locale/en/docs/guides/security/index.md

```diff
@@ -0,0 +1,427 @@
+dependencies.
+* Carefully check the _package.json_ file for errors/typos in the names of the
+dependencies.
+
+### Memory Access Violation (CWE-284)
+
+Memory-based or heap-based attacks depends on a combination of memory management
+errors and an exploitable memory allocator.
+Like all runtimes, Node.js is vulnerable to these attacks if your projects runs
+on a shared machine.
+Using a secure heap is useful for preventing sensitive information from leaking
+due to pointer overruns and underruns.
+
+Unfortunately, secure heap is not available on Windows.
+More information can be found on Node.js [secure-heap documentation][].
+
+**Mitigation**
+
+* Use `--secure-heap=n` depending on your application where _n_ is the allocated
+maximum byte size.
+* Do not run your production app in a shared machine.
+
+### Monkey Patching (CWE-349)
+
+Monkey patching refers to the modification of properties in runtime aiming to
+change the existing behavior. Example:
+
+```js
+// eslint-disable-next-line no-extend-native
+Array.prototype.push = function (item) {
+  // overriding the global [].push
+};
+```
+
+This is **not** considered a vulnerability, it falls under the following
+statement: _“Node.js trusts the code being run”_.
+
+**Mitigation**
+
+The `--frozen-intrinsics` flag enables **experimental** frozen intrinsics, which
+means all the built-in JavaScript objects and functions are recursively frozen.
+Therefore, the following snippet **will not** override the default behavior of
+`Array.prototype.push`
+
+```js
+// eslint-disable-next-line no-extend-native
+Array.prototype.push = function (item) {
+  // overriding the global [].push
+};
+
+// Uncaught:
+// TypeError <Object <Object <[Object: null prototype] {}>>>:
+// Cannot assign // to read only property 'push' of object ''
+```
+
+However, it’s important to mention you can still define new globals and replace
+existing globals using `globalThis`
+```console
+> globalThis.foo = 3; foo; // you can still define new globals
+3
+> globalThis.Array = 4; Array; // However, you can also replace existing globals
+4
+```
+
+### Prototype Pollution Attacks (CWE-1321)
+
+Prototype pollution refers to the possibility to modify or inject properties
+into Javascript language items by abusing the usage of _\_proto\__,
+_constructor_, and _prototype_.
+
+Example:
+
+```js
+const a = {"a": 1, "b": 2};
+const data = JSON.parse('{"__proto__": { "polluted": true}}');
+
+const c = Object.assign({}, a, data);
+console.log(c.polluted); // true
+```
+
+This is considered a potential vulnerability inherited from the JavaScript
+language.
+
+**Examples**:
+
+* [CVE-2022-21824][] (Node.js)
+* [CVE-2018-3721][] (3rd Party library: Lodash)
+
+**Mitigation**
+
+* Avoid [insecure recursive merges][], see [CVE-2018-16487][].
+* Implement JSON Schema validations for external/untrusted requests.
+* Create Objects without prototype by using `Object.create(null)`.
+* Freezing the prototype: `Object.freeze(MyObject.prototype)`.
```

Comment:
might also mention `--disable-proto` since most stuff can come from there in real world CVEs.